app/src/main/java/com/example/reduxtestapp/data/repository/country/CountryRepositoryImpl.kt

```diff
@@ -38,22 +63,53 @@ class CountryRepositoryImpl (
                 countriesApi.getAllCountries()
             }
         }.mapLeft { throwable ->
-            when (throwable) {
-                is HttpException -> {
-                    if (throwable.code() == 404) {
-                        ErrorState.EmptyListError
-                    } else {
-                        ErrorState.CountriesError(throwable.message).also {
-                            logError(throwable.message)
-                        }
-                    }
-                }
-                else -> {
-                    ErrorState.CountriesError(throwable.message).also {
-                        logError(throwable.message)
-                    }
-                }
+            handleError(throwable)
+        }
+    }
+
+    override suspend fun searchByLanguageAndCurrency(
+        language: String,
+        currency: String
+    ): Either<ErrorState, List<CountryDto>> {
+        return either {
+            val deferredLanguageResult = CoroutineScope(Dispatchers.IO).async {
+                searchByLanguage(language)
+            }
+            val deferredCurrencyResult = CoroutineScope(Dispatchers.IO).async {
+                searchByCurrency(currency)
             }
```

Comment:
check out `withContext` block and replace `CoroutineScope(Dispatchers.IO)` with one `withContext`